### PR TITLE
[LOG4J2-3426] Refactors wrapping code

### PR DESCRIPTION
**Pull request by ppkarwasz**
Refactors the wrapping logic into single `wrap` or `adapt` methods to be sure we always perform the correct logic.

### CODE REVIEW

File: log4j-1.2-api/src/main/java/org/apache/log4j/builders/appender/FileAppenderBuilder.java
Comment (ppkarwasz) [inline log4j-1.2-api/src/main/java/org/apache/log4j/builders/appender/FileAppenderBuilder.java:137]: This and similar changes actually modify the function's behavior: if the argument of `AppenderWrapper.wrap` is `null`, then `null` is returned. However this should be counteracted by #768 .

File: log4j-1.2-api/src/main/java/org/apache/log4j/Category.java
Comment (ppkarwasz) [inline log4j-1.2-api/src/main/java/org/apache/log4j/Category.java:379]: This is the main change: if `coreAppender` is an `AppenderAdapter` and **no** filters are configured, then the wrapped object is returned.

@garydgregory: can you check if this is Ok?

**Comment by pooja-achievement** [comment]
@ppkarwasz,

Hi Piotr,

I am not building log4j2 1.x bridge jars but downloading the jar. I wanted to test with this Pull Request changes, is it possible for you to provide log4j1.x bridge.jar log4j2-core.jar and log4j2-api.jar built with this change ??

**Comment by ppkarwasz** [comment]
Hi @pooja-achievement ,
I am waiting for a second opinion on these changes: there is not much sense in testing something that will not end up in the next release. If everything is alright, we can generate a snapshot.

File: log4j-1.2-api/src/main/java/org/apache/log4j/bridge/FilterWrapper.java
Comment (garydgregory) [inline log4j-1.2-api/src/main/java/org/apache/log4j/bridge/FilterWrapper.java:31]: Test if the input is already a FilterWrapper and return it?

File: log4j-1.2-api/src/main/java/org/apache/log4j/bridge/LayoutAdapter.java
Comment (garydgregory) [inline log4j-1.2-api/src/main/java/org/apache/log4j/bridge/LayoutAdapter.java:33]: Test if input is already a LayoutAdapter and return it?

File: log4j-1.2-api/src/main/java/org/apache/log4j/bridge/LayoutWrapper.java
Comment (garydgregory) [inline log4j-1.2-api/src/main/java/org/apache/log4j/bridge/LayoutWrapper.java:30]: Test if input is already a LayoutWrapper and return it?

File: log4j-1.2-api/src/main/java/org/apache/log4j/builders/appender/SyslogAppenderBuilder.java
Comment (garydgregory) [inline log4j-1.2-api/src/main/java/org/apache/log4j/builders/appender/SyslogAppenderBuilder.java:134]: I like the existing formatting better.

**Review by garydgregory — CHANGES_REQUESTED** [review]
- I like the refactoring in general, but:
- Let's stick to one term: adapt instead of wrap, since sometimes, we "unwap" or do nothing.
- Add some minimal Javadoc to new public methods to help maintainers.
- Adapter methods should check if the input is already "adapted" and return it for consistency and avoid multi-layer wrapping by accident.
- See my comments.

**Review by garydgregory — APPROVED** [review]
LGTM.

**Pull request merged**
